v0/destinations/kafka/transform.js

```diff
@@ -1,36 +1,74 @@
+/* eslint-disable no-restricted-syntax */
+const cloneDeep = require("lodash/cloneDeep");
 const { getIntegrationsObj } = require("../../util");
 
 function batch(destEvents) {
   const respList = [];
-  const batchedRequest = [];
-  const metadata = [];
-  destEvents.forEach(event => {
-    metadata.push(event.metadata);
-    batchedRequest.push(event.message);
-  });
-  respList.push({
-    batchedRequest,
-    metadata,
-    destination: destEvents[0].destination
-  });
+
+  // Grouping the events by topic
+  const groupedEvents = destEvents.reduce((acc, event) => {
```

Comment:
is there lodash groupBy functionality we can leverage ?

Reply:
Yes addressed